### ISSUE TITLE
"ingest:git:..." fails if aimed at a submodule with relocated data dir

### ISSUE THREAD

**Issue by warpfork** [polydawn/reach#16]
tl;dr if the git repo you try to ingest has a `.git` *file* instead of the full *dir* there, it fails.

Which is fairly common behavior for submodules in recent versions of git (anything newer than ~2015ish, iirc) -- the git data dir is relocated under `.git/modules` of the *parent* repo, and a `.git` *file* in the submodule points towards that.

Apparently `git.PlainOpen(pathstr)` (at least at our currently linked version of the relevant libraries) doesn't account for this case...?

(This may be tangentially related to, or involve the same knowledge as, https://github.com/polydawn/rio/issues/9 -- the essence of the issue is we really need to take a long hard look at all the 'open' functions this library provides and figure out which ones do the right things.  Or at worst, if it so happens that none of them do and we need to push some patches upstream.)
